test/labeling/model/test_label_model.py

```diff
@@ -88,11 +100,155 @@ def test_augmented_L_construction(self):
+
+        label_model = self._set_up_model(L, deps=[(1, 2)])
+        true_mask = np.array(
+            [
+                [0, 0, 1, 1, 1, 1],
+                [0, 0, 1, 1, 1, 1],
+                [1, 1, 0, 0, 0, 0],
+                [1, 1, 0, 0, 0, 0],
+                [1, 1, 0, 0, 0, 0],
+                [1, 1, 0, 0, 0, 0],
+            ]
+        )
+
+        mask = label_model.mask.numpy()
+        np.testing.assert_array_equal(mask, true_mask)
+
+    def test_init_params(self):
+        L = np.array([[1, 2, 1], [1, 0, 1]])
+        label_model = self._set_up_model(L, class_balance=[0.6, 0.4])
+
+        mu_init = label_model.mu_init.numpy()
+        true_mu_init = np.array(
+            [[1.0, 0.0], [0.0, 0.0], [0.0, 0.0], [0.0, 0.8750], [1.0, 0.0], [0.0, 0.0]]
+        )
+        np.testing.assert_array_equal(mu_init, true_mu_init)
+
+        label_model._set_class_balance(class_balance=[0.3, 0.7], Y_dev=None)
+        label_model._init_params()
+
+        mu_init = label_model.mu_init.numpy()
+        true_mu_init = np.array(
+            [[1.0, 0.0], [0.0, 0.0], [0.0, 0.0], [0.0, 0.5], [1.0, 0.0], [0.0, 0.0]]
+        )
+        np.testing.assert_array_equal(mu_init, true_mu_init)
+
+    def test_predict_proba(self):
+        L = np.array([[1, 2, 1], [1, 2, 1]])
+        label_model = self._set_up_model(L)
+
+        label_model.mu = nn.Parameter(label_model.mu_init.clone())
+        probs = label_model.predict_proba(L)
+
+        true_probs = np.array([[0.99, 0.01], [0.99, 0.01]])
+        np.testing.assert_array_almost_equal(probs, true_probs)
+
+        L = np.array([[1, 2, 1], [1, 2, 0]])
+        label_model = self._set_up_model(L, deps=[(1, 2)])
+        label_model._get_augmented_label_matrix(L, higher_order=True)
+
+        label_model.mu = nn.Parameter(label_model.mu_init.clone())
+        probs = label_model.predict_proba(L)
+
+        true_probs = np.array([0.99, 0.01])
+        np.testing.assert_array_almost_equal(probs[1, :], true_probs)
+
+    def test_loss(self):
+        L = np.array([[1, 0, 1], [1, 2, 0]])
+        label_model = self._set_up_model(L)
+        label_model._get_augmented_label_matrix(L, higher_order=True)
+
+        label_model.mu = nn.Parameter(label_model.mu_init.clone() + 0.05)
+        self.assertAlmostEqual(
+            label_model.loss_l2(l2=1.0).detach().numpy().ravel()[0], 0.03
+        )
+        self.assertAlmostEqual(
+            label_model.loss_l2(l2=np.ones(6)).detach().numpy().ravel()[0], 0.03
+        )
+        self.assertAlmostEqual(
+            label_model.loss_mu().detach().numpy().ravel()[0], 0.6751751
+        )
+
+    def test_inv_loss(self):
+        L = np.array([[1, 0, 1], [1, 2, 1]])
+        label_model = self._set_up_model(L, deps=[(1, 2)])
+        label_model.inv_form = True
+
+        label_model.Z = nn.Parameter(torch.ones(label_model.d, label_model.k)).float()
+
+        Q = label_model.get_Q()
+        self.assertAlmostEqual(Q[0, 0], 0.89285714)
```

Comment:
Fewer decimals here? Did you compute this by hand or just see what an execution of `get_Q()` spat out?

Reply:
I got it from Mathematica after doing the needful operations on the label matrix and substituting variables. I'll change to check 2 decimal places